bgpd/bgp_community.c

```diff
@@ -990,13 +1011,33 @@ void bgp_remove_community_from_aggregate(struct bgp_aggregate *aggregate,
 						aggr_community);
 			community_free(&ret_comm);
 
-			community_free(&aggregate->community);
+			bgp_compute_aggregate_community_val(aggregate);
+		}
+	}
+}
+
+void bgp_remove_comm_from_aggregate_hash(struct bgp_aggregate *aggregate,
+		struct community *community)
+{
+
+	struct community *aggr_community = NULL;
+	struct community *ret_comm = NULL;
+
+	if ((!aggregate)
+	    || (!aggregate->community_hash)
+	    || (!community))
+		return;
+
+	/* Look-up the community in the hash.
+	 */
+	aggr_community = bgp_aggr_community_lookup(aggregate, community);
+	if (aggr_community) {
+		aggr_community->refcnt--;
 
-			/* Compute aggregate's community.
-			 */
-			hash_iterate(aggregate->community_hash,
-				     bgp_aggr_community_prepare,
-				     &aggregate->community);
+		if (aggr_community->refcnt == 0) {
+			ret_comm = hash_release(aggregate->community_hash,
```

Comment:
is this needed really? looks like you do not check it below, just freeing.

Reply:
Yes it's required.

Reply:
I mean `ret_comm`, not `hash_release`.

Reply:
Yes bucket->data needs to free. So it's required.